File: Javascript/Data Structures/Stack.js
/*class Stack {
  constructor(item) {
    this._stack = [];
    if (item)
      this._stack.push(item);
    }
  push(item) {
    return this
      ._stack
      .push(item);
  }
  pop(item) {
    return this
      ._stack
      .pop(item);
  }
  list() {
    for (const item of this._stack.reverse()) {
      console.log(item);
    }
  }
  top() {
    return this._stack[this._stack.length-1];
  }
}*/

function Stack(item) {
  this._stack = [];
  if (item)
    this._stack.push(item);
  
  this.push = (item) => this
    ._stack
    .push(item);
  this.pop = (item) => this
    ._stack
    .pop(item);
  this.list = () => this
    ._stack
    .reduce((str, val) => `${val} ${str}`);

  this.top = () => this._stack[this._stack.length - 1];
}

module.exports = Stack;

File: Python/Data Structures/stack.py
#!/usr/bin/python

class Stack:
    def __init__(self):
        self.stack = []
        self.count = 0

    def push(self, item):
        self.stack.append(item)
        self.count += 1

    def pop(self):
        self.count -= 1
        return self.stack.pop()

    def last(self):
        if self.count is 0:
            return None
        else:
            return self.stack[-1]

    def list(self):
        # Returns 'stack' order, i.e. LIFO
        return self.stack[::-1]
